### PR TITLE
Issue #7624: Updated doc for NoWhitespaceAfterCheck

### PR DESCRIPTION
**Pull request by Shaman20**
Fixes issue for NoWhitespaceAfterCheck #7624 

image

### CODE REVIEW

**Comment by Shaman20** [comment]
image

**Comment by Shaman20** [comment]
Website After Updating

image

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java [hunk 1]
```diff
@@ -99,6 +99,28 @@
  *   &lt;property name=&quot;allowLineBreaks&quot; value=&quot;false&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:103]: Please add same example under default config to show how it works  by default and how violations are changing after configuration change

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java (hunk 1 above)
Comment (Shaman20) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:125]: I have added the same example under default config

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java [hunk 2]
```diff
@@ -99,6 +121,28 @@
...
+ * class Test {
+ *   int a, b;
+ *   Test(int a, int b) {
+ *     this.
+ *         a = a; // violation, linebreak after '.' is not allowed
+ *     this.b = b; // OK
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:131]: Now use will confused, why config is changing, but violations are not.
Please make java examples to in such content to have on some line violation with one config and no violation with another.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java (hunk 2 above)
Comment (Shaman20) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java]: done

**Comment by Shaman20** [comment]
CLI exceution

image

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java (hunk 2 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:121]: Please use same java code in all examples.
To let user see how config changes result in violation changes on same code

**Comment by romani** [comment]
See comment above

**Comment by Shaman20** [comment]
Website Image

image

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java (hunk 2 above)
Comment (Shaman20) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:121]: CLI Exceution for default config

image

**Comment by Shaman20** [comment]
Could you please tell me what does this CI error mean. I didn't understand it looked like I had some extra characters somewhere I'm not sure about it.

image

**Comment by romani** [comment]
GitHub, generate website

**Comment by Shaman20** [comment]
> GitHub, generate website

Do you mean to generate whitespaces webpage for the changes made?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java (hunk 2 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:128]: Too much focus on violation after DOT, please add few more violation(+2 of different kind) of different kind.
Check do space validation for bunch of tokens 
https://checkstyle.org/config_whitespace.html#NoWhitespaceAfter_Properties

File: src/xdocs/config_whitespace.xml [hunk 3]
```diff
@@ -1424,6 +1442,28 @@ import static java.math.BigInteger.ZERO;
...
   &lt;property name=&quot;allowLineBreaks&quot; value=&quot;false&quot;/&gt;
 &lt;/module&gt;
+        </source>
+        <p>Example:</p>
+        <source>
+class Test {
+  int a, b;
+  Test(int a, int b) {
+    this.
+        a = a; // violation, linebreak after '.' is not allowed
```
Comment (romani) [inline src/xdocs/config_whitespace.xml:1452]: I really mean that java code has to be identical.
The difference should be only in trailing comments.

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by Shaman20** [comment]
> > Could you please tell me what does
> 
> At the top you see "spellchecker" it demands good names.
> 
> Please remove from PR dependency-reduced-pom.xml file or add it to .gitignore list in separate commit of this PR.

Should I change my method name and variable names?

**Comment by romani** [comment]
Yes , do good names, user like good names.

**Comment by Shaman20** [comment]
Website

image

File: src/xdocs/config_whitespace.xml (hunk 3 above)
Comment (Shaman20) [reply src/xdocs/config_whitespace.xml]: done

File: .gitignore [hunk 4]
```diff
@@ -23,7 +23,6 @@ checkstyle.iml
 checkstyle.ipr
 checkstyle.iws
 .idea
-
```
Comment (romani) [inline .gitignore:26]: Remove this change

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java [hunk 5]
```diff
@@ -99,6 +119,26 @@
  *   &lt;property name=&quot;allowLineBreaks&quot; value=&quot;false&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class Test {
+ *
+ *   public void dotOperator(String s) {
+ *     Integer.parseInt(s); // Ok
+ *     Integer. parseInt(s); // violation, '.' is followed by whitespace
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:128]: Add some code to show allowLineBreaks in action.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .gitignore (hunk 4 above)
Comment (Shaman20) [reply .gitignore:26]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java (hunk 5 above)
Comment (Shaman20) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:140]: done

File: .gitignore (hunk 4 above)
Comment (romani) [reply .gitignore:26]: this file should not have any changes , please do

**Comment by romani** [comment]
last item, see above

File: .gitignore (hunk 4 above)
Comment (Shaman20) [reply .gitignore:26]: Should I remove it and keep the line empty or delete the line?

Comment (romani) [reply .gitignore]: revert all changes from this file.
keep it untouched

Comment (Shaman20) [reply .gitignore]: done

**Comment by romani** [comment]
Github, generate website

**Comment by Shaman20** [comment]
> Github, generate website

image


image

**Comment by romani** [comment]
@Shaman20 , this comment trigger Github bot to generate website and share a link.

**Comment by Shaman20** [comment]
> @Shaman20 , this comment trigger Github bot to generate website and share a link. you do not need to do anything, just wait for several minutes.

Oh ok my bad.

File: src/xdocs/config_whitespace.xml [hunk 6]
```diff
@@ -1446,6 +1471,32 @@ import static java.math.BigInteger.ZERO;
...
+  }
+
+  public void bitwiseNot(int a) {
+    a = ~ a; // Ok
+    a = ~a; // Ok
+  }
+}
         </source>
         <p>
          If the annotation is between the type and the array, the check will skip validation for
```
Comment (romani) [inline src/xdocs/config_whitespace.xml:1502]: please do me a favor and move this paragraph
to be after  this paragraph:
```
will be ignored.
        </p>
```

examples should be config + code. This note paragraph is misplaced.

**Review by romani — CHANGES_REQUESTED** [review]
very last item:

File: src/xdocs/config_whitespace.xml (hunk 6 above)
Comment (Shaman20) [reply src/xdocs/config_whitespace.xml:1502]: > please do me a favor and move this paragraph to be after this paragraph:
> 
> ```
> will be ignored.
>         </p>
> ```
> 
> examples should be config + code. This note paragraph is misplaced.

Sorry to say this, but I didn't get what you asked me to do could you please elaborate?..

Comment (Shaman20) [reply src/xdocs/config_whitespace.xml:1502]: <p>
         If the annotation is between the type and the array, the check will skip validation for

Should I put the above entire paragraph after this?

<a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#INDEX_OP">
                INDEX_OP</a>
          will be ignored.
        </p>

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java [hunk 7]
```diff
@@ -91,6 +91,32 @@
  * <pre>
  * &lt;module name=&quot;NoWhitespaceAfter&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class Test {
+ *
+ *    public void lineBreak(String x) {
+ *     Integer.
+ *     parseInt(x); // Ok
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java:100]: add 2 spaces to indent this line to right, to show that it is line wrap of previos line.

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: src/xdocs/config_whitespace.xml (hunk 6 above)
Comment (Shaman20) [reply src/xdocs/config_whitespace.xml:1502]: done

**Comment by romani** [comment]
Github, generate website

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by romani** [comment]
Github, generate website

**Review by romani — APPROVED** [review]
Ok to merge

File: src/xdocs/config_whitespace.xml [hunk 8]
```diff
@@ -1437,7 +1444,32 @@ import static java.math.BigInteger.ZERO;
         <source>
 &lt;module name=&quot;NoWhitespaceAfter&quot;/&gt;
         </source>
+        <p>Example:</p>
+        <source>
+class Test {
 
+   public void lineBreak(String x) {
```
Comment (romani) [inline src/xdocs/config_whitespace.xml:1451]: Please remove extra space at the beginning of this line to align indentation

**Review by romani — CHANGES_REQUESTED** [review]
Items 

Clear to see at https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/6744d64_2022004823/config_whitespace.html#NoWhitespaceAfter_Examples

File: src/xdocs/config_whitespace.xml (hunk 8 above)
Comment (Shaman20) [reply src/xdocs/config_whitespace.xml]: done

**Comment by Shaman20** [comment]
Hey I am interested in devops and so far I have contributed to two gsoc first issues. So what kind of issues/label should I look for to contribute to devops related?

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**